### PR TITLE
resolve dependencies from additional repositories

### PR DESCRIPTION
**Pull request by mpollmeier**
* also support and document credentials for repo authentication
* allow to specify multi-value config options without error-prone comma-separation: change for all config parameters
* make non-optimistic, i.e. fail on error

### CODE REVIEW

File: core/src/main/scala/replpp/Dependencies.scala [hunk 1]
```diff
@@ -1,31 +1,56 @@
...
+    Try {
+      val parseResults = RepositoryParser.repositories(additionalRepositories)
+      if (parseResults.isSuccess) {
+        parseResults.either.getOrElse(???)
+      } else {
+        val failures = parseResults.either.left.getOrElse(???)
+        throw new AssertionError(s"error while trying to parse given repository coordinates: ${failures.mkString(",")}")
       }
     }
+  }
```
Comment (maltek) [inline core/src/main/scala/replpp/Dependencies.scala:37]:
```suggestion
    Try {
      val parseResults = RepositoryParser.repositories(additionalRepositories)
      parseResults.either match {
        case Right(res) => res
        case Left(failures) =>
          throw new AssertionError(s"error while trying to parse given repository coordinates: ${failures.mkString(",")}")
      }
    }
  }
```

File: core/src/test/scala/replpp/DependenciesTests.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    errorMessage should include("error while trying to parse given repository coordinates")
+  }
+}
+
+object DependenciesTests {
+
+  def main(args: Array[String]): Unit = {
+    /** only to be run manually, sorry...
+      * verify that we can access an artifact that's only available on a separate, password-protected repository
+      * note: relies on the local ~/config/coursier/credentials.properties
```
Comment (maltek) [inline core/src/test/scala/replpp/DependenciesTests.scala:54]:
```suggestion
      * note: relies on the local ~/.config/coursier/credentials.properties
```

**Review by maltek — APPROVED** [review]
:chefs_kiss:

**Comment by mpollmeier** [comment]
good suggestions, thank you!

**Pull request merged**